File: packages/core/dto/clusterrolebinding.go
package dto

type CRBSubject struct {
	Kind      string
	Name      string
	Namespace string
}

type ClusterRoleBindingRef struct {
	Name string `json:"name"`
}

type ClusterRoleBinding struct {
	Name          string
	Bindings      string
	Age           string
	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	RoleRefKind   string
	RoleRefName   string
	RoleRefGroup  string
	Subjects      []CRBSubject
}

File: packages/core/frontend/src/types/resources/clusterrolebinding.ts
import type { ManagedField } from "./shared";

export interface CRBSubject {
  Kind: string;
  Name: string;
  Namespace: string;
}

export interface ClusterRoleBinding {
  Name: string;
  Bindings: string;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  RoleRefKind: string;
  RoleRefName: string;
  RoleRefGroup: string;
  Subjects: CRBSubject[];
}
